pkg/manager/internal.go

```diff
@@ -260,6 +263,20 @@ func (cm *controllerManager) SetFields(i interface{}) error {
 	return nil
 }
 
+// AddMetricsExtraHandler adds extra handler served on path to the http server that serves metrics.
+func (cm *controllerManager) AddMetricsExtraHandler(path string, handler http.Handler) {
```

Comment:
I'd return an error here instead of adding the panic

Reply:
Yep, done